### ISSUE TITLE
SVG - stop-opacity cuts the color completely

### ISSUE THREAD

**Issue by mgrudzinska** [Samsung/thorvg#521]
```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 140 140">
  <defs>
    <linearGradient
       id="grad" x1="0" y1="0" x2="0.2" y2="0.2" spreadMethod="reflect">
      <stop style="stop-color:#ff0000;stop-opacity:1;" offset="0"/>
      <stop style="stop-color:#0000ff;stop-opacity:0;" offset="1"/>
    </linearGradient>
  </defs>
  <rect x="0" y="0" width="100" height="100" fill="url(#grad)"/>
</svg>
```

tvg result:
![5before](https://user-images.githubusercontent.com/67589014/123713776-6d369900-d875-11eb-9151-e732c13a78b7.PNG)

expected result:
![5expected](https://user-images.githubusercontent.com/67589014/123713795-76276a80-d875-11eb-9501-f4d8b55163ce.PNG)

**Comment by hermet** [comment]
@mgrudzinska can you explain this issue more detail? so that other volunteers can assign it.

**Comment by mgrudzinska** [comment]
sorry, I edited the descr

**Comment by mmaciola** [comment]
note it is thorvg issue, not just svg
```
void tvgDrawCmds(tvg::Canvas* canvas)
{
    if (!canvas) return;

    //Background
    auto shape = tvg::Shape::gen();
    shape->appendRect(0, 0, WIDTH, HEIGHT, 0, 0);
    shape->fill(255, 255, 255, 255);
    canvas->push(move(shape));

    //Shape
    auto shape1 = tvg::Shape::gen();
    shape1->appendRect(0, 0, WIDTH, HEIGHT, 0, 0);

    //LinearGradient
    tvg::Fill::ColorStop colorStops[2];
    colorStops[0] = {0, 255, 0, 0, 255};
    colorStops[1] = {1, 0, 0, 255, 25};

    auto fill = tvg::LinearGradient::gen();
    fill->linear(0, 0, WIDTH, HEIGHT);
    fill->colorStops(colorStops, 2);

    shape1->fill(move(fill));
    if (canvas->push(move(shape1)) != tvg::Result::Success) return;
}
```
![Screenshot from 2021-06-29 17-46-03](https://user-images.githubusercontent.com/71131832/123828638-feebe800-d901-11eb-94c6-3840503ab6c4.png)

**Comment by mgrudzinska** [comment]
solved in #531

**Comment by hermet** [comment]
@mgrudzinska  reverted that patch since it has a regression bug. Let's revisit it once stabilizing.
https://github.com/Samsung/thorvg/commit/e1eabf7dc3b74b1739e0938d5090609db4636f1a

**Comment by mgrudzinska** [comment]
solved in #580
